src/web/client/services/copresenceWorker.js

```diff
@@ -0,0 +1,161 @@
+            const { container, services } = await azureClient.getContainer(
+                id,
+                containerSchema
+            );
+            if (container.connectionState !== ConnectionState.Connected) {
+                await new Promise((resolve) => {
+                    container.once("connected", () => {
+                        resolve();
+                    });
+                });
+            }
+            this._container = container;
+            this._audience = services.audience;
+            this._userSharedMap = container.initialObjects.sharedState;
+        }
+        return {
+            container: this._container,
+            audience: this._audience,
+            map: this._userSharedMap,
+        };
+    }
+}
+
+async function loadContainer(config, id, swpId, file) {
+    try {
+        const { container, audience, map } =
+            await AzureFluidClient.fetchContainerAndService(config, swpId);
+
+        const existingMembers = audience.getMembers();
+
+        const myself = audience.getMyself();
+
+        const currentUser = {
+            containerId: id,
+            fileName: file.fileName,
+            filePath: file.filePath,
+            userName: myself.userName,
+        };
+
+        map.set(myself.userId, currentUser);
+
+        audience.on("memberRemoved", (clientId, member) => {
+
+            if (!existingMembers.get(member.userId)) {
+                self.postMessage({
+                    type: "member-removed",
+                    userId: member.userId,
+                });
+            }
+        });
+
+        if (!initial) {
+            existingMembers.forEach(async (value, key) => {
+                const otherUser = value;
+
+                self.postMessage({
+                    type: "client-data",
+                    userName: otherUser.userName,
+                    userId: key,
+                    containerId: swpId,
+                    fileName: otherUser.fileName === undefined ? "On Studio" : otherUser.fileName,
+                    filePath: otherUser.filePath,
+                });
+            });
+            initial = true;
+        }
+
+        map.on("valueChanged", async (changed, local) => {
+            if (!local) {
+                const otherUser = map.get(changed.key);
+                // eslint-disable-next-line no-undef
+                await self.postMessage({
+                    type: "client-data",
+                    userId: changed.key,
+                    userName: otherUser.userName,
+                    containerId: swpId,
+                    fileName: otherUser.fileName,
+                    filePath: otherUser.filePath,
+                });
+            }
+        });
+    } catch (error) {
+        console.log(`Error retrieving container: ${error}`);
+    }
+}
+
+function runFluidApp() {
+    // Listen for messages from the extension
+    // eslint-disable-next-line no-undef
+    self.addEventListener("message", async (event) => {
```

Comment:
In follow-up PR we should revisit what all telemetry we need to log form this script.